rocAL/source/readers/image/cifar10_data_reader.cpp

```diff
@@ -171,17 +226,53 @@ Reader::Status CIFAR10DataReader::subfolder_reading() {
     if (!_file_names.empty())
         LOG("CIFAR10DataReader  Total of " + TOSTR(_file_names.size()) + " images loaded from " + _full_path)
 
+    auto dataset_size = _file_count_all_shards;
+    size_t padded_samples = 0;
+    // Pad the _file_names with last element of the shard in the vector when _pad_last_batch_repeated is True
+    padded_samples = ((_shard_size > 0) ? _shard_size : largest_shard_size_without_padding()) % _batch_size;
+    _last_batch_padded_size = (_batch_size > 1) ? (_batch_size - padded_samples) : 0;
+
+    if (_pad_last_batch_repeated == true) { 
+        // pad the last sample when the dataset_size is not divisible by
+        // the number of shard's (or) when the shard's size is not
+        // divisible by the batch size making each shard having equal
+        // number of samples
+        for (uint shard_id = 0; shard_id < _shard_count; shard_id++) {
+            uint start_idx = (dataset_size * shard_id) / _shard_count;
+            uint actual_shard_size_without_padding = std::floor((shard_id + 1) * dataset_size / _shard_count) - floor(shard_id * dataset_size / _shard_count);
+            uint largest_shard_size = std::ceil(dataset_size * 1.0 / _shard_count);
+            auto start = _file_names.begin() + start_idx;
+            auto end = _file_names.begin() + start_idx + actual_shard_size_without_padding;
+            auto start_offset = _file_offsets.begin() + start_idx;
+            auto end_offset = _file_offsets.begin() + start_idx + actual_shard_size_without_padding;
+            auto start_file_idx = _file_idx.begin() + start_idx;
+            auto end_file_idx = _file_idx.begin() + start_idx + actual_shard_size_without_padding;
+            if (start != end && start <= _file_names.end() &&
+                end <= _file_names.end()) {
+                _all_shard_file_names_padded.insert(_all_shard_file_names_padded.end(), start, end);
+                _all_shard_file_offsets.insert(_all_shard_file_offsets.end(), start_offset, end_offset);
+                _all_shard_file_idxs.insert(_all_shard_file_idxs.end(), start_file_idx, end_file_idx);
+            }
+            if (largest_shard_size % _batch_size) {
+                size_t num_padded_samples = 0;
+                num_padded_samples = (largest_shard_size - actual_shard_size_without_padding) + _batch_size - (largest_shard_size % _batch_size);
+                _file_count_all_shards += num_padded_samples;
+                _all_shard_file_names_padded.insert(_all_shard_file_names_padded.end(), num_padded_samples, _all_shard_file_names_padded.back());
```

Comment:
Are all these _all_shard_* variables required?
Or can we use the existing ones itself like file source reader please check